resources/scripts.js

```diff
@@ -125,112 +171,267 @@ function buttonPosition(loc) {
+        return;
+    }
+    if (playerMoney > lastPlayerMoney) {
+        playerMoneyElement.style.transition = 'all 0.5s';
+        playerMoneyElement.style.fontSize = '175%';
+        playerMoneyElement.style.color = '#0000ff';
+        await new Promise(resolve => setTimeout(resolve, 500));
+        playerMoneyElement.style.transition = 'all 0.5s';
+        playerMoneyElement.style.fontSize = '120%';
+        playerMoneyElement.style.color = '#00ff00';
+        return;
+    }
+    return;
 }
 
 function displayMessage(str) {
     document.getElementById('current-message').innerHTML = str;
     const toastBootstrap = bootstrap.Toast.getOrCreateInstance(messageTrigger);
     toastBootstrap.show();
+    return;
 }
 
-function boardClick(e) { // To implement: advanced betting during game
-    document.getElementById('clickNum').innerHTML = e.target.alt;
-    let i = parseInt(e.target.alt);
+function boardClick(e) {
+    let numClicked = parseInt(e.target.alt);
+    const PLACENOPOINT = `You cannot place on ${numClicked} because no point is open`
+    if (pointOpen === false) {
+        displayMessage(PLACENOPOINT);
+        return;
+    }
+    if (numClicked === playerPoint) {
+        displayMessage(PLACEONPOINT);
+        return;
+    }
+    makePlace(numClicked);
+    return;
 }
 
-function validateBet() {
+function validateBet(betAmount) {
     if (isNaN(betAmount) || betAmount > playerMoney || betAmount < 1) {
-        displayMessage('Please insert valid bet');
+        displayMessage(INVALIDBET);
         betAmount = 0;
         return false;
     }
     return true;
 }
 
-function diceRoll() {
-    const dieOne = getRandomInt(1,7); 
-    const dieTwo = getRandomInt(1,7); 
+async function diceRoll() {
+    const dieOne = getRandomInt(1, 7);
+    const dieTwo = getRandomInt(1, 7);
     playerRoll = dieOne + dieTwo;
-    document.getElementById('show-roll').innerHTML = `${dieOne} + ${dieTwo} = ${playerRoll}`;
+    firstDieElement.setAttribute('src', `images/dice_animated.gif`);
+    secondDieElement.setAttribute('src', `images/dice_animated.gif`);
+    await new Promise(resolve => setTimeout(resolve, 1000));
+    firstDieElement.setAttribute('src', `images/dice_${dieOne}.png`);
+    secondDieElement.setAttribute('src', `images/dice_${dieTwo}.png`);
+    showRollElement.innerHTML = `${dieOne} + ${dieTwo} = ${playerRoll}`;
+    rollDisplayElement.innerHTML = `${playerRoll}`;
+    return;
 }
 
 function getRandomInt(min, max) {
-  const minCeiled = Math.ceil(min);
-  const maxFloored = Math.floor(max);
-  return Math.floor(Math.random() * (maxFloored - minCeiled) + minCeiled);
+    const minCeiled = Math.ceil(min);
+    const maxFloored = Math.floor(max);
+    return Math.floor(Math.random() * (maxFloored - minCeiled) + minCeiled);
 }
 
 let playerMoney = 100;
-let playerRoll = 0;
+let lastPlayerMoney = 0;
+let playerRoll = 1;
 let pointOpen = false;
 let playerPoint = 0;
 let betAmount = 0;
+let comeAmount = 0;
```

Comment:
Could the scope of comeAmount and BetAmount possibly be narrowed to their respective functions?

Reply:
I believe `comeAmount` can be narrowed to its respective function, but I'm not sure if `betAmount` can

Reply:
Made variables function specific.